internal/broker/instance_create.go

```diff
@@ -263,11 +263,16 @@ func (b *ProvisionEndpoint) validateAndExtract(details domain.ProvisionDetails,
 		return ersContext, parameters, err
 	}
 
-	if !b.config.AllowModulesParameters {
+	if !b.config.AllowModulesParameters && parameters.Modules != nil {
 		b.log.Infof("modules section passed to API, but AllowModulesParameters is set to false. Parameters will be reset to nil")
 		parameters.Modules = nil
 	}
 
+	if parameters.Modules != nil && parameters.Modules.Default == nil && parameters.Modules.List != nil && len(parameters.Modules.List) == 0 {
```

Comment:
It's relatively hard to grasp logical expression.
Could we add a comment or extract logical value using descriptive name?